### PR TITLE
[crypto/ec] don't assume points are of order group->order (1.1.0)

### PR DESCRIPTION
**Pull request by romen**
This is a backport of #6535 to 1.1.0

~~~It's a WIP because I need feedback on 2 things:~~~

- ~~~I'm cherry picking for backport #6190 , so if this gets backported after approval there this needs to be rebased~~~
- ~~~https://github.com/openssl/openssl/pull/6535#issuecomment-398998193 @bernd-edlinger does this mean you prefer I don't backport the negative test?~~~

**It's now ready for review.**

##### Checklist
- [x] tests are added or updated

### CODE REVIEW

**Comment by romen** [comment]
Ping @levitte @paulidale @kroeckx as they reviewed the original PR

**Comment by levitte** [comment]
I'm fine with this as it currently stands.  Will approve as soon as it's out of WIP (unless there are new changes to review).

**Comment by romen** [comment]
So I'll just wait for @bernd-edlinger to comment on including the tests in `evptests.txt`.
If there is no objection to keeping things as they currently are in this commit I'll remove the WIP.

**Comment by bernd-edlinger** [comment]
no, I don't mind either way.

File: test/evptests.txt [hunk 1]
```diff
@@ -19144,6 +19144,35 @@ PeerKey=KAS-ECC-CDH_B-571_C24-Peer-PUBLIC
 Ctrl=ecdh_cofactor_mode:1
 SharedSecret=02da266a269bdc8d8b2a0c6bb5762f102fc801c8d5394a9271539136bd81d4b69cfbb7525cd0a983fb7f7e9deec583b8f8e574c6184b2d79831ec770649e484dc006fa35b0bffd0b
 
+# for cofactor-order points, CECDH should fail. Test that.
```
Comment (bernd-edlinger) [inline test/evptests.txt]: could you update this comment to match the latest version of the original PR?

Comment (romen) [reply test/evptests.txt]: will do!

**Comment by levitte** [comment]
Merged.

7b3e775a6a78650bbd3e8e19a5aa12981880402b [crypto/ec] don't assume points are of order group->order
cc39f9250957dfe6e9f1b62a4eca1863e8451483 ec/ec_mult.c: get BN_CTX_start,end sequence right.

**Pull request closed without merging**